src/libsyntax/tokenstream.rs

```diff
@@ -115,6 +136,16 @@ impl TokenTree {
     }
 }
 
+impl<CTX> HashStable<CTX> for TokenStream
+    where CTX: crate::StableHashingContextLike
+{
+    fn hash_stable(&self, hcx: &mut CTX, hasher: &mut StableHasher) {
+        for sub_tt in self.trees() {
```

Comment:
@nnethercote It seems like this iterator does a bunch of `Lrc` clones. Is there some reason this can't just hash the `Vec<TreeAndJoint>`? It also doesn't hash `IsJoint`, which seems questionable.

Reply:
@petrochenkov Do you know why this ignores `IsJoint`?